src/include/txn_inline.h

```diff
@@ -972,6 +975,34 @@ __wt_txn_read_upd_list_internal(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt,
             continue;
 
         WT_ORDERED_READ(prepare_state, upd->prepare_state);
+
+        /*
+         * We previously found a prepared update, check if the update has the same transaction id,
+         * if it does it must not be visible as it is part of the same transaction as the previous
+         * prepared update.
+         */
+        if (prepare_txnid != WT_TXN_NONE && upd->txnid == prepare_txnid) {
+            /*
+             * If we see an update with prepare resolved this indicates that the read, which is
+             * configured to ignore prepared updates raced with the commit of the same prepared
+             * transaction. Increment a stat to track this.
+             *
+             * This case exists as reconciliation chooses which update to read to disk in a newest
+             * to oldest fashion, and cannot handle prepared update resolution happening in the same
+             * newest to oldest direction. Instead prepared update resolution is recursively done
+             * from oldest to newest. Which mean that our reader would see a prepared update
+             * followed by a committed update.
+             *
+             * FIXME-WT-11195: There is an alternate solution which would have reconciliation forget
```

Comment:
I think the comment here is worthwhile, outlining an alternative solution, but I wouldn't reference a FIXME. It isn't an obviously better solution, and I suspect we won't schedule the work.

Reply:
Really? I thought of it the other way round, we could leave an alternative solution in the comments but without a fix me it will never get done. Though you are probably correct that we wouldn't schedule the work any time soon I'd hope we'd get to it in the distant future?

Reply:
I think of it like this: I'm not convinced it's a better solution - but it would be useful to leave a breadcrumb if we find a problem with the current solution.

Reply:
I can remove the fix me and WT-11195 will be left as future work, sounds good?

Reply:
3fc1bfe0e